D3D12Bundles/D3D12Bundles.cs

```diff
@@ -187,45 +178,42 @@ struct Vertex {
         /// Render the scene
         /// </summary>
         public virtual void OnRender() {
-            using (var pe = new ProfilingEvent(mCommandQueue, "Render")) {
+            using (var pe = new ProfilingEvent(mGraphicsDevice.DirectCommandQueue.NativeCommandQueue, "Render")) {
                 // Record all the commands we need to render the scene into the command list.
-                PopulateCommandList(mCurrentFrameResource);
+                CompiledCommandList compiledCommandList = PopulateCommandList(mGraphicsDevice.CommandList, mCurrentFrameResource);
 
                 // Execute the command list.
-                mCommandQueue.ExecuteCommandList(mCommandList);
+                mGraphicsDevice.DirectCommandQueue.ExecuteCommandList(compiledCommandList);
             }
 
             // Present and update the frame index for the next frame.
-            mSwapChain.Present(1, 0).CheckError();
-            mFrameIndex = mSwapChain.CurrentBackBufferIndex;
+            mPresenter.Present();
 
             // Signal and increment the fence value.
-            mCurrentFrameResource.mFenceValue = mFenceValue;
-            mCommandQueue.Signal(mFence, mFenceValue).CheckError();
-            mFenceValue++;
+            mCurrentFrameResource.mFenceValue = mGraphicsDevice.DirectCommandQueue.AddSignal();
         }
 
         public virtual void OnDestroy() {
             // Ensure that the GPU is no longer referencing resources that are about to be
             // cleaned up by the destructor.
             {
-                var fence = mFenceValue;
-                var lastCompletedFence = mFence.CompletedValue;
-
-                // Signal and increment the fence value.
-                mCommandQueue.Signal(mFence, mFenceValue).CheckError();
-                mFenceValue++;
+                // Add signal to the command queue.
+                var fence = mGraphicsDevice.DirectCommandQueue.AddSignal();
 
                 // Wait until the previous frame is finished.
-                if (lastCompletedFence < fence) {
-                    mFence.SetEventOnCompletion(fence, mFenceEvent).CheckError();
-                    mFenceEvent.WaitOne();
-                }
+                mGraphicsDevice.DirectCommandQueue.WaitForSignal(fence);
             }
 
             foreach (var frameResource in mFrameResources) {
                 frameResource.Dispose();
             }
+
+#if DEBUG
+            if (DXGI.DXGIGetDebugInterface1(out Vortice.DXGI.Debug.IDXGIDebug1? dxgiDebug).Success) {
+                dxgiDebug!.ReportLiveObjects(DXGI.DebugAll, Vortice.DXGI.Debug.ReportLiveObjectFlags.Summary | Vortice.DXGI.Debug.ReportLiveObjectFlags.IgnoreInternal);
+                dxgiDebug.Dispose();
+            }
+#endif
```

Comment:
Additional logging added on shutdown, if native debugging is enabled and a we're in a debug build.  Copied from D3D12Mutiny class.